software/nosql/src/main/java/brooklyn/entity/nosql/redis/RedisClusterImpl.java

```diff
@@ -72,12 +120,12 @@ public void restart() {
         throw new UnsupportedOperationException();
     }
 
-    protected boolean calculateServiceUp() {
-        boolean up = false;
-        for (Entity member : slaves.getMembers()) {
-            if (Boolean.TRUE.equals(member.getAttribute(SERVICE_UP))) up = true;
-        }
-        return up;
-    }
+    private boolean calculateServiceUp() {
+        if (!Boolean.TRUE.equals(getMaster().getAttribute(SERVICE_UP))) return false;
 
+        for (Entity member : getSlaves().getMembers())
```

Comment:
Also can replace the manual checks with enricher (in `initEnrichers`)
```
ServiceStateLogic.newEnricherFromChildrenUp().checkChildrenOnly().requireUpChildren(QuorumChecks.all()).addTo(this);
```

Where `QuorumChecks` is in `brooklyn.util.collections.QuorumCheck`.